test/jdk/javax/swing/SwingWorker/TestDoneBeforeDoInBackground.java

```diff
@@ -27,20 +27,23 @@
  *           before the 'doInBackground' is finished
  * @run main TestDoneBeforeDoInBackground
  */
+import java.beans.PropertyChangeListener;
+import java.beans.PropertyChangeEvent;
 import javax.swing.SwingWorker;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 public class TestDoneBeforeDoInBackground {
 
-    private static boolean doInBackground = false;
+    private static AtomicBoolean doInBackgroundStarted = new AtomicBoolean(false);
+    private static AtomicBoolean doInBackgroundFinished = new AtomicBoolean(false);
     private static final int WAIT_TIME = 200;
     private static final long CLEANUP_TIME = 1000;
     private static final CountDownLatch doneLatch = new CountDownLatch(1);
```

Comment:
```suggestion
    private static final int WAIT_TIME = 200;
    private static final long CLEANUP_TIME = 1000;

    private static final AtomicBoolean doInBackgroundStarted = new AtomicBoolean(false);
    private static final AtomicBoolean doInBackgroundFinished = new AtomicBoolean(false);
    private static final CountDownLatch doneLatch = new CountDownLatch(1);
```

Separate the real constants from the objects which track the state. The flags should also be marked `final`.